hplsql/src/main/java/org/apache/hive/hplsql/Expression.java

```diff
@@ -565,7 +575,11 @@ public void operatorConcatSql(HplsqlParser.Expr_concatContext ctx) {
     sql.append("CONCAT(");
     int cnt = ctx.expr_concat_item().size();
     for (int i = 0; i < cnt; i++) {
-      sql.append(evalPop(ctx.expr_concat_item(i)).toString());
+      String concatStr = evalPop(ctx.expr_concat_item(i)).toString();
+      if (!concatStr.startsWith("'")) {
```

Comment:
AFAIK `concat` expects string parameters. The actual parameters can be a
* string literals which needs quotation 
* or an string typed expressions which should not be quoted
Could you please share an example when quotation is missing but required at this point?

Reply:
For example below example
`SELECT '#' || TRIM(' Hello ') || '#';`
Here HPLSQL internally calls concat and when `TRIM(' Hello ')` is evaluated it will become `Hello` without quotes so here concat will throw error. To avoid it if the concat item is not starting with single quote then enclosing it with single quotes.

Reply:
Fixed